src/y-websocket.js

```diff
@@ -180,8 +198,9 @@ export class WebsocketProvider extends Observable {
    * @param {Object<string,string>} [opts.params]
    * @param {typeof WebSocket} [opts.WebSocketPolyfill] Optionall provide a WebSocket polyfill
    * @param {number} [opts.resyncInterval] Request server state every `resyncInterval` milliseconds
+   * @param {reconnectPredicate} [opts.reconnectPredicate] Takes number of attempts to reconnect - return true/false to continue trying
```

Comment:
I feel it would be more ideomatic to create a parameter `maxReconnectAttempts` instead of a function. Once `maxReconnectAttempts` is reached, the wsProvider should emit `status: "connection-error", description: "max reconnect attempts"`.

A function would also be fine if you find use for it. But then the parameter of `reconnectPredicate` should be the provider instance instead. The second parameter can be the current number of attempts.